### ISSUE TITLE
Parser code uses old openapi2 "definitions" instead of "components/schemas"

### ISSUE THREAD

**Issue by tommyjcarpenter** [koxudaxi/datamodel-code-generator#895]
I am debugging this package, which explodes parsing a remote reference openapi. 

```
Traceback (most recent call last):
  File "/Users/tommy/.pyenv/versions/3.10.6/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/datamodel_code_generator/__main__.py", line 574, in main
    generate(
  File "/Users/tommy/.pyenv/versions/3.10.6/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/datamodel_code_generator/__init__.py", line 371, in generate
    results = parser.parse()
  File "/Users/tommy/.pyenv/versions/3.10.6/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/datamodel_code_generator/parser/base.py", line 460, in parse
    self.parse_raw()
  File "/Users/tommy/.pyenv/versions/3.10.6/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/datamodel_code_generator/parser/openapi.py", line 397, in parse_raw
    self.parse_raw_obj(
  File "/Users/tommy/.pyenv/versions/3.10.6/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1180, in parse_raw_obj
    self.parse_obj(name, JsonSchemaObject.parse_obj(raw), path)
  File "/Users/tommy/.pyenv/versions/3.10.6/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1200, in parse_obj
    self.parse_ref(obj, path)
  File "/Users/tommy/.pyenv/versions/3.10.6/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1141, in parse_ref
    self.parse_ref(property_value, path)
  File "/Users/tommy/.pyenv/versions/3.10.6/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1120, in parse_ref
    self.resolve_ref(obj.ref)
  File "/Users/tommy/.pyenv/versions/3.10.6/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1109, in resolve_ref
    self._parse_file(
  File "/Users/tommy/.pyenv/versions/3.10.6/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1324, in _parse_file
    models = get_model_by_path(raw, object_paths)
  File "/Users/tommy/.pyenv/versions/3.10.6/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/datamodel_code_generator/parser/jsonschema.py", line 54, in get_model_by_path
    return get_model_by_path(schema[keys[0]], keys[1:])
KeyError: ''
```

I am sure I am on the latest version:

```
/Users/tommy/.pyenv/versions/3.10.6/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/datamodel_code_generator/parser/jsonschema.py", line 54, in get_model_by_path
```

When I open this file, I see lots of mentions of `#/definitions`, but thats openapi2 lingo as noted at the bottom of the page here: https://swagger.io/docs/specification/components/

```
OpenAPI 2.0                    OpenAPI 3.0
'#/definitions/User'         → '#/components/schemas/User'
```

However, there are 0 usages of `#/Components`

Does this package support OpenAPI3, or am I invoking it wrong? I used:

```
datamodel-codegen --input-file-type openapi --input dataapi/openapi.yaml --output model.py
``` 

DEBUGGING: 

The remote reference file we are trying to parse is actually public, here: `http://merq-api-assets.s3-website-us-east-1.amazonaws.com/swagger_components.yaml`

the `paths` is showing up as:
```
['', 'merq-api-assets.s3-website-us-east-1.amazonaws.com', 'swagger_components.yaml#', 'components', 'schemas', 'AirflowConfig']
```

However, ` 'swagger_components.yaml#', 'components',` does not look right

**Comment by tommyjcarpenter** [comment]
Our remote reference notation:

```
- $ref: "http://merq-api-assets.s3-website-us-east-1.amazonaws.com/swagger_components.yaml#/components/parameters/uuidParam"
```

is in the exact format exampled here: https://swagger.io/docs/specification/components/

```
$ref: 'https://api.example.com/v2/openapi.yaml#/components/schemas/User'
```

**Comment by tommyjcarpenter** [comment]
I also tried renaming it to the standard `openapi.yaml` rather than `swagger_components.yaml` but that wasnt it.

**Comment by tommyjcarpenter** [comment]
OK! I got this to work with the following diff:

**Comment by tommyjcarpenter** [comment]
Im confused as to whether this project supports openapi3 or I am really doing something wrong, becuase as shown at that link, openapi3 eliminated `definitions` in favor of `components/schemas`

**Comment by tommyjcarpenter** [comment]
(I was unable to make a pull request into this repo)

**Comment by koxudaxi** [comment]
@tommyjcarpenter
Thank you for creating the PR.
Oh, I don't expect a remote file on OpenAPI :(
I will fix it.

**Comment by tommyjcarpenter** [comment]
@koxudaxi let me know if you need our original spec but that swagger documentation shows it (and we heavily make use of remote references to DRY our specs eg two different openapis share a few common objects)

**Comment by koxudaxi** [comment]
The code generator doesn't support OpenAPI 2(a.k.a Swagger).
we should update `parser/openapi.py` for OpenAPI. 
I will request to review it with you after I create the PR.

**Comment by tommyjcarpenter** [comment]
@koxudaxi sorry I used the word "Swagger" but im really talking about OpenAPI 3. 
the code i showed in my PR is from openapi2 - `definitions` is gone

**Comment by koxudaxi** [comment]
@tommyjcarpenter 
I have released the new version as `0.13.5` :tada:
